# Introduction
See https://github.com/wtsi-team112/fits/issues/4

This notebook examines discrepancies between an initial Pf6.2 manifest created by Magnus from fits which has 23,542 files, and one created by Richard which has 17,001.

In [1]:
%run ../setup.ipynb

python 3.5.2 |Continuum Analytics, Inc.| (default, Jul  2 2016, 17:53:06) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]
numpy 1.13.3
scipy 0.19.1
pandas 0.22.0
numexpr 2.6.4
pysam 0.8.4
pysamstats 0.24.3
petlx 1.0.3
vcf 0.6.8
h5py 2.7.0
tables 3.4.2
zarr 2.2.0
scikit-allel 1.1.10


In [2]:
richard_fn = '/nfs/team112_internal/rp7/src/github/wtsi-team112/Pipelines_issues/work/65_Pf_6_2_Pv4_like_manifest/Pf_62_Pv4_like_manifest_20180906.txt'
magnus_fn = './Pf_62.12-09-2018'

# Read in data

In [3]:
pd.options.display.max_columns = 30
df_richard = pd.read_csv(richard_fn, sep='\t', index_col=6)
print(df_richard.shape)
df_richard[0:3]

(17001, 26)


path  \
irods_path                                                                         
/seq/1276/1276_5_nonhuman.bam  /lustre/scratch118/malaria/team112/pipelines/s...   
/seq/1276/1276_6_nonhuman.bam  /lustre/scratch118/malaria/team112/pipelines/s...   
/seq/1333/1333_8_nonhuman.bam  /lustre/scratch118/malaria/team112/pipelines/s...   

                                           study    sample             lane  \
irods_path                                                                    
/seq/1276/1276_5_nonhuman.bam  1001-PF-ML-DJIMDE  PM0001-C  1276_5_nonhuman   
/seq/1276/1276_6_nonhuman.bam  1001-PF-ML-DJIMDE  PM0001-C  1276_6_nonhuman   
/seq/1333/1333_8_nonhuman.bam  1001-PF-ML-DJIMDE  PM0001-C  1333_8_nonhuman   

                               reads  paired sanger_sample_id  taxon_id  \
irods_path                                                                
/seq/1276/1276_5_nonhuman.bam     -1       1              NaN      5833   
/seq/1276/1276_6_nonhuman.bam     -1       1              NaN      5833   
/seq/1333/1333_8_nonhuman.bam     -1       1              NaN      5833   

                               study_lims  \
irods_path                                  
/seq/1276/1276_5_nonhuman.bam         578   
/seq/1276/1276_6_nonhuman.bam         578   
/seq/1333/1333_8_nonhuman.bam         578   

                                                             study_name  \
irods_path                                                                
/seq/1276/1276_5_nonhuman.bam  Plasmodium falciparum genome variation 1   
/seq/1276/1276_6_nonhuman.bam  Plasmodium falciparum genome variation 1   
/seq/1333/1333_8_nonhuman.bam  Plasmodium falciparum genome variation 1   

                               id_run  position  tag_index qc_complete  \
irods_path                                                               
/seq/1276/1276_5_nonhuman.bam    1276         5        NaN  2008-09-24   
/seq/1276/1276_6_nonhuman.bam    1276         6        NaN  2008-09-24   
/seq/1333/1333_8_nonhuman.bam    1333         8        NaN  2008-09-24   

                               manual_qc description instrument_name  \
irods_path                                                             
/seq/1276/1276_5_nonhuman.bam        NaN         NaN             NaN   
/seq/1276/1276_6_nonhuman.bam        NaN         NaN             NaN   
/seq/1333/1333_8_nonhuman.bam        NaN         NaN             NaN   

                              instrument_model  forward_read_length  \
irods_path                                                            
/seq/1276/1276_5_nonhuman.bam            HiSeq                  100   
/seq/1276/1276_6_nonhuman.bam            HiSeq                  100   
/seq/1333/1333_8_nonhuman.bam            HiSeq                  100   

                               requested_insert_size_from  \
irods_path                                                  
/seq/1276/1276_5_nonhuman.bam                         NaN   
/seq/1276/1276_6_nonhuman.bam                         NaN   
/seq/1333/1333_8_nonhuman.bam                         NaN   

                               requested_insert_size_to  human_percent_mapped  \
irods_path                                                                      
/seq/1276/1276_5_nonhuman.bam                       NaN                   NaN   
/seq/1276/1276_6_nonhuman.bam                       NaN                   NaN   
/seq/1333/1333_8_nonhuman.bam                       NaN                   NaN   

                                 subtrack_filename  subtrack_files_bytes  \
irods_path                                                                 
/seq/1276/1276_5_nonhuman.bam  1276_5_nonhuman.srf          5.464048e+09   
/seq/1276/1276_6_nonhuman.bam  1276_6_nonhuman.srf          5.479595e+09   
/seq/1333/1333_8_nonhuman.bam  1333_8_nonhuman.srf          5.136273e+09   

                              ebi_run_acc previous_release  
irods_path                             

In [4]:
df_magnus = pd.read_csv(magnus_fn, index_col=6)
print(df_magnus.shape)
df_magnus[0:3]

(23542, 26)


path  \
irods_path                                                                   
/seq/8563/8563_8#49.bam  /lustre/scratch118/malaria/team112/pipelines/s...   
/seq/8563/8563_8#50.bam  /lustre/scratch118/malaria/team112/pipelines/s...   
/seq/8563/8563_8#51.bam  /lustre/scratch118/malaria/team112/pipelines/s...   

                                           study    sample       lane  \
irods_path                                                              
/seq/8563/8563_8#49.bam  1017-PF-GH-AMENGA-ETEGO  PF0386-C  8563_8#49   
/seq/8563/8563_8#50.bam        1001-PF-ML-DJIMDE  PM0078-C  8563_8#50   
/seq/8563/8563_8#51.bam  1017-PF-GH-AMENGA-ETEGO  PF0397-C  8563_8#51   

                              reads  paired sanger_sample_id  taxon_id  \
irods_path                                                               
/seq/8563/8563_8#49.bam  31300752.0     1.0         PF0386-C      5833   
/seq/8563/8563_8#50.bam  18722556.0     1.0         PM0078-C      5833   
/seq/8563/8563_8#51.bam  35463112.0     1.0         PF0397-C      5833   

                         study_lims                                study_name  \
irods_path                                                                      
/seq/8563/8563_8#49.bam       578.0  Plasmodium falciparum genome variation 1   
/seq/8563/8563_8#50.bam       578.0  Plasmodium falciparum genome variation 1   
/seq/8563/8563_8#51.bam       578.0  Plasmodium falciparum genome variation 1   

                         id_run  position  tag_index qc_complete  manual_qc  \
irods_path                                                                    
/seq/8563/8563_8#49.bam  8563.0       8.0       49.0         NaN        1.0   
/seq/8563/8563_8#50.bam  8563.0       8.0       50.0         NaN        1.0   
/seq/8563/8563_8#51.bam  8563.0       8.0       51.0         NaN        1.0   

                        description instrument_name instrument_model  \
irods_path                                                             
/seq/8563/8563_8#49.bam         NaN            HS24            HiSeq   
/seq/8563/8563_8#50.bam         NaN            HS24              NaN   
/seq/8563/8563_8#51.bam         NaN            HS24              NaN   

                         forward_read_length  requested_insert_size_from  \
irods_path                                                                 
/seq/8563/8563_8#49.bam                100.0                       200.0   
/seq/8563/8563_8#50.bam                100.0                       200.0   
/seq/8563/8563_8#51.bam                100.0                       200.0   

                         requested_insert_size_to human_percent_mapped  \
irods_path                                                               
/seq/8563/8563_8#49.bam                     300.0                 TODO   
/seq/8563/8563_8#50.bam                     300.0                 TODO   
/seq/8563/8563_8#51.bam                     300.0                 TODO   

                        subtrack_filename subtrack_files_bytes ebi_run_acc  \
irods_path                                                                   
/seq/8563/8563_8#49.bam              TODO                 TODO   ERR216455   
/seq/8563/8563_8#50.bam              TODO                 TODO   ERR216456   
/seq/8563/8563_8#51.bam              TODO                 TODO   ERR216457   

                        previous_release  
irods_path                                
/seq/8563/8563_8#49.bam          UNKNOWN  
/seq/8563/8563_8#50.bam          UNKNOWN  
/seq/8563/8563_8#51.bam          UNKNOWN

# Do files contain unique lanelets?

In [5]:
print(df_richard['lane'].size, df_richard['lane'].nunique())

17001 17001


In [6]:
print(df_magnus['lane'].size, df_magnus['lane'].nunique())

23542 23530


In [7]:
df_magnus.loc[df_magnus['lane'].duplicated(keep=False), ['study', 'sample', 'lane', 'reads', 'paired']]

study    sample        lane  \
irods_path                                                                   
/seq/10868/10868_3#1.cram          1083-PF-GH-CONWAY  PF0668-C   10868_3#1   
/seq/10868/10868_3#2.cram          1083-PF-GH-CONWAY  PF0669-C   10868_3#2   
/seq/10868/10868_3#3.cram          1083-PF-GH-CONWAY  PF0673-C   10868_3#3   
/seq/10868/10868_3#4.cram          1083-PF-GH-CONWAY  PF0674-C   10868_3#4   
/seq/10868/10868_3#5.cram          1083-PF-GH-CONWAY  PF0675-C   10868_3#5   
/seq/10868/10868_3#6.cram          1083-PF-GH-CONWAY  PF0687-C   10868_3#6   
/seq/10868/10868_3#7.cram          1083-PF-GH-CONWAY  PF0693-C   10868_3#7   
/seq/10868/10868_3#8.cram          1083-PF-GH-CONWAY  PF0696-C   10868_3#8   
/seq/10868/10868_3#9.cram          1083-PF-GH-CONWAY  PF0701-C   10868_3#9   
/seq/10868/10868_3#10.cram         1083-PF-GH-CONWAY  PF0704-C  10868_3#10   
/seq/10868/10868_3#11.cram         1083-PF-GH-CONWAY  PF0706-C  10868_3#11   
/seq/10868/10868_3#12.cram         1083-PF-GH-CONWAY  PF0712-C  10868_3#12   
/seq/10868/no_cal/10868_3#1.cram   1083-PF-GH-CONWAY  PF0668-C   10868_3#1   
/seq/10868/no_cal/10868_3#10.cram  1083-PF-GH-CONWAY  PF0704-C  10868_3#10   
/seq/10868/no_cal/10868_3#11.cram  1083-PF-GH-CONWAY  PF0706-C  10868_3#11   
/seq/10868/no_cal/10868_3#12.cram  1083-PF-GH-CONWAY  PF0712-C  10868_3#12   
/seq/10868/no_cal/10868_3#2.cram   1083-PF-GH-CONWAY  PF0669-C   10868_3#2   
/seq/10868/no_cal/10868_3#3.cram   1083-PF-GH-CONWAY  PF0673-C   10868_3#3   
/seq/10868/no_cal/10868_3#4.cram   1083-PF-GH-CONWAY  PF0674-C   10868_3#4   
/seq/10868/no_cal/10868_3#5.cram   1083-PF-GH-CONWAY  PF0675-C   10868_3#5   
/seq/10868/no_cal/10868_3#6.cram   1083-PF-GH-CONWAY  PF0687-C   10868_3#6   
/seq/10868/no_cal/10868_3#7.cram   1083-PF-GH-CONWAY  PF0693-C   10868_3#7   
/seq/10868/no_cal/10868_3#8.cram   1083-PF-GH-CONWAY  PF0696-C   10868_3#8   
/seq/10868/no_cal/10868_3#9.cram   1083-PF-GH-CONWAY  PF0701-C   10868_3#9   

                                        reads  paired  
irods_path                                             
/seq/10868/10868_3#1.cram          42478110.0     1.0  
/seq/10868/10868_3#2.cram          36974150.0     1.0  
/seq/10868/10868_3#3.cram          30283304.0     1.0  
/seq/10868/10868_3#4.cram          36744668.0     1.0  
/seq/10868/10868_3#5.cram          35900900.0     1.0  
/seq/10868/10868_3#6.cram          20195200.0     1.0  
/seq/10868/10868_3#7.cram          40069810.0     1.0  
/seq/10868/10868_3#8.cram          44304020.0     1.0  
/seq/10868/10868_3#9.cram          44454962.0     1.0  
/seq/10868/10868_3#10.cram         33727438.0     1.0  
/seq/10868/10868_3#11.cram         41042808.0     1.0  
/seq/10868/10868_3#12.cram         38882374.0     1.0  
/seq/10868/no_cal/10868_3#1.cram          NaN     NaN  
/seq/10868/no_cal/10868_3#10.cram         NaN     NaN  
/seq/10868/no_cal/10868_3#11.cram         NaN     NaN  
/seq/10868/no_cal/10868_3#12.cram         NaN     NaN  
/seq/10868/no_cal/10868_3#2.cram          NaN     NaN  
/seq/10868/no_cal/10868_3#3.cram          NaN     NaN  
/seq/10868/no_cal/10868_3#4.cram          NaN     NaN  
/seq/10868/no_cal/10868_3#5.cram          NaN     NaN  
/seq/10868/no_cal/10868_3#6.cram          NaN     NaN  
/seq/10868/no_cal/10868_3#7.cram          NaN     NaN  
/seq/10868/no_cal/10868_3#8.cram          NaN     NaN  
/seq/10868/no_cal/10868_3#9.cram          NaN     NaN

In [8]:
df_magnus.loc[df_magnus.index.to_series().str.contains('no_cal')].shape

(12, 26)

### Summary
Magnus file contains 12 lanelets which are duplicated, with one version containing 'no_cal' in file path

# Lanelets in richard but not magnus

In [9]:
df_richard_not_magnus = df_richard.loc[~df_richard.index.to_series().isin(df_magnus.index)]
print(df_richard_not_magnus.shape)
df_richard_not_magnus[0:5]

(2385, 26)


path  \
irods_path                                                                         
/seq/1276/1276_5_nonhuman.bam  /lustre/scratch118/malaria/team112/pipelines/s...   
/seq/1276/1276_6_nonhuman.bam  /lustre/scratch118/malaria/team112/pipelines/s...   
/seq/1333/1333_8_nonhuman.bam  /lustre/scratch118/malaria/team112/pipelines/s...   
/seq/1276/1276_7_nonhuman.bam  /lustre/scratch118/malaria/team112/pipelines/s...   
/seq/1276/1276_8_nonhuman.bam  /lustre/scratch118/malaria/team112/pipelines/s...   

                                           study    sample             lane  \
irods_path                                                                    
/seq/1276/1276_5_nonhuman.bam  1001-PF-ML-DJIMDE  PM0001-C  1276_5_nonhuman   
/seq/1276/1276_6_nonhuman.bam  1001-PF-ML-DJIMDE  PM0001-C  1276_6_nonhuman   
/seq/1333/1333_8_nonhuman.bam  1001-PF-ML-DJIMDE  PM0001-C  1333_8_nonhuman   
/seq/1276/1276_7_nonhuman.bam  1001-PF-ML-DJIMDE  PM0002-C  1276_7_nonhuman   
/seq/1276/1276_8_nonhuman.bam  1001-PF-ML-DJIMDE  PM0002-C  1276_8_nonhuman   

                               reads  paired sanger_sample_id  taxon_id  \
irods_path                                                                
/seq/1276/1276_5_nonhuman.bam     -1       1              NaN      5833   
/seq/1276/1276_6_nonhuman.bam     -1       1              NaN      5833   
/seq/1333/1333_8_nonhuman.bam     -1       1              NaN      5833   
/seq/1276/1276_7_nonhuman.bam     -1       1              NaN      5833   
/seq/1276/1276_8_nonhuman.bam     -1       1              NaN      5833   

                               study_lims  \
irods_path                                  
/seq/1276/1276_5_nonhuman.bam         578   
/seq/1276/1276_6_nonhuman.bam         578   
/seq/1333/1333_8_nonhuman.bam         578   
/seq/1276/1276_7_nonhuman.bam         578   
/seq/1276/1276_8_nonhuman.bam         578   

                                                             study_name  \
irods_path                                                                
/seq/1276/1276_5_nonhuman.bam  Plasmodium falciparum genome variation 1   
/seq/1276/1276_6_nonhuman.bam  Plasmodium falciparum genome variation 1   
/seq/1333/1333_8_nonhuman.bam  Plasmodium falciparum genome variation 1   
/seq/1276/1276_7_nonhuman.bam  Plasmodium falciparum genome variation 1   
/seq/1276/1276_8_nonhuman.bam  Plasmodium falciparum genome variation 1   

                               id_run  position  tag_index qc_complete  \
irods_path                                                               
/seq/1276/1276_5_nonhuman.bam    1276         5        NaN  2008-09-24   
/seq/1276/1276_6_nonhuman.bam    1276         6        NaN  2008-09-24   
/seq/1333/1333_8_nonhuman.bam    1333         8        NaN  2008-09-24   
/seq/1276/1276_7_nonhuman.bam    1276         7        NaN  2008-09-24   
/seq/1276/1276_8_nonhuman.bam    1276         8        NaN  2008-09-24   

                               manual_qc description instrument_name  \
irods_path                                                             
/seq/1276/1276_5_nonhuman.bam        NaN         NaN             NaN   
/seq/1276/1276_6_nonhuman.bam        NaN         NaN             NaN   
/seq/1333/1333_8_nonhuman.bam        NaN         NaN             NaN   
/seq/1276/1276_7_nonhuman.bam        NaN         NaN             NaN   
/seq/1276/1276_8_nonhuman.bam        NaN         NaN             NaN   

                              instrument_model  forward_read_length  \
irods_path                                                            
/seq/1276/1276_5_nonhuman.bam            HiSeq                  100   
/seq/1276/1276_6_nonhuman.bam            HiSeq                  100   
/seq/1333/1333_8_nonhuman.bam            HiSeq                  100   
/seq/1276/1276_7_nonhuman.bam            HiSeq                  100   
/seq/1276/1276_8_nonhuman.bam            HiSeq                  100   

                               requested_in

In [10]:
df_richard_not_magnus['sanger_sample_id'].isnull().value_counts()

True     2167
False     218
Name: sanger_sample_id, dtype: int64

In [11]:
df_richard_not_magnus['previous_release'].value_counts(dropna=False)

Pf6.0    2383
Pf6.1       2
Name: previous_release, dtype: int64

In [12]:
df_richard_not_magnus['study'].value_counts(dropna=False)

1017-PF-GH-AMENGA-ETEGO          299
1044-PF-KH-FAIRHURST             251
1001-PF-ML-DJIMDE                186
1031-PF-SEA-PLOWE                179
1010-PF-TH-ANDERSON              134
1026-PF-GN-CONWAY                132
1004-PF-BF-OUEDRAOGO             128
1011-PF-KH-SU                    110
1016-PF-TH-NOSTEN                 92
1147-PF-MR-CONWAY                 86
1015-PF-KE-NZILA                  83
1006-PF-GM-CONWAY                 75
1155-PF-ID-PRICE                  70
1021-PF-PG-MUELLER                60
1043-PF-GB-RAYNER                 58
1022-PF-MW-OCHOLLA                53
1052-PF-TRAC-WHITE                44
1032-PF-BRHN-SMITHEE              42
1041-PF-US-FERDIG                 40
1037-PF-GM-CONWAY                 31
1012-PF-KH-WHITE                  27
1042-PF-HN-RANFORD-CARTWRIGHT     22
1013-PF-PEGB-BRANCH               21
1098-PF-ET-GOLASSA                19
1023-PF-CO-ECHEVERRI-GARCIA       17
1020-PF-VN-BONI                   16
1014-PF-SSA-SUTHERLAND            16
1

In [13]:
df_richard_not_magnus['id_run'].value_counts(dropna=False).head(10)

6345     84
6263     78
6466     74
6295     72
6222     69
17234    60
6936     46
6901     45
6007     39
17397    38
Name: id_run, dtype: int64

In [14]:
df_magnus.loc[df_magnus['id_run'] == 6345.0]

Empty DataFrame
Columns: [path, study, sample, lane, reads, paired, sanger_sample_id, taxon_id, study_lims, study_name, id_run, position, tag_index, qc_complete, manual_qc, description, instrument_name, instrument_model, forward_read_length, requested_insert_size_from, requested_insert_size_to, human_percent_mapped, subtrack_filename, subtrack_files_bytes, ebi_run_acc, previous_release]
Index: []

In [15]:
df_magnus.loc[df_magnus['id_run'] == 6263.0]

Empty DataFrame
Columns: [path, study, sample, lane, reads, paired, sanger_sample_id, taxon_id, study_lims, study_name, id_run, position, tag_index, qc_complete, manual_qc, description, instrument_name, instrument_model, forward_read_length, requested_insert_size_from, requested_insert_size_to, human_percent_mapped, subtrack_filename, subtrack_files_bytes, ebi_run_acc, previous_release]
Index: []

In [16]:
df_magnus.loc[df_magnus['id_run'] == 6466.0]

Empty DataFrame
Columns: [path, study, sample, lane, reads, paired, sanger_sample_id, taxon_id, study_lims, study_name, id_run, position, tag_index, qc_complete, manual_qc, description, instrument_name, instrument_model, forward_read_length, requested_insert_size_from, requested_insert_size_to, human_percent_mapped, subtrack_filename, subtrack_files_bytes, ebi_run_acc, previous_release]
Index: []

In [17]:
df_magnus.loc[df_magnus['id_run'] == 17234.0].shape

(10, 26)

In [18]:
df_richard.loc[df_richard['id_run'] == 17234.0].shape

(70, 26)

### Summary
There are 2,385 files in Richard's manifest, which aren't in Magnus's file. All of these were in previous releases so clearly should be present. I could see no obvious pattern in terms of studies or runs which were missing. For run 17234 for example, Magnus file had 10 lanelets, whereas Richard's had 70. The majority of the differences had no sanger_sample_id.

# Lanelets in magnus but not richard

In [19]:
df_magnus_not_richard = df_magnus.loc[~df_magnus.index.to_series().isin(df_richard.index)]
print(df_magnus_not_richard.shape)
df_magnus_not_richard[0:5]

(8926, 26)


path  \
irods_path                                                                     
/seq/23384/23384_1#1.cram  /lustre/scratch118/malaria/team112/pipelines/s...   
/seq/23384/23384_1#2.cram  /lustre/scratch118/malaria/team112/pipelines/s...   
/seq/23384/23384_1#3.cram  /lustre/scratch118/malaria/team112/pipelines/s...   
/seq/23384/23384_1#4.cram  /lustre/scratch118/malaria/team112/pipelines/s...   
/seq/23384/23384_1#5.cram  /lustre/scratch118/malaria/team112/pipelines/s...   

                          study    sample       lane       reads  paired  \
irods_path                                                                 
/seq/23384/23384_1#1.cram   NaN  SPT09376  23384_1#1  30821088.0     1.0   
/seq/23384/23384_1#2.cram   NaN  SPT09379  23384_1#2  38921494.0     1.0   
/seq/23384/23384_1#3.cram   NaN  SPT09398  23384_1#3  31690686.0     1.0   
/seq/23384/23384_1#4.cram   NaN  SPT09396  23384_1#4  29677456.0     1.0   
/seq/23384/23384_1#5.cram   NaN  SPT09413  23384_1#5  30322306.0     1.0   

                          sanger_sample_id  taxon_id  study_lims  \
irods_path                                                         
/seq/23384/23384_1#1.cram  4660STDY6750851      5833      4660.0   
/seq/23384/23384_1#2.cram  4660STDY6750859      5833      4660.0   
/seq/23384/23384_1#3.cram  4660STDY6750868      5833      4660.0   
/seq/23384/23384_1#4.cram              NaN      5833         NaN   
/seq/23384/23384_1#5.cram  4660STDY6750861      5833      4660.0   

                               study_name   id_run  position  tag_index  \
irods_path                                                                
/seq/23384/23384_1#1.cram  1204-PF-GM-CP1  23384.0       1.0        1.0   
/seq/23384/23384_1#2.cram  1204-PF-GM-CP1  23384.0       1.0        2.0   
/seq/23384/23384_1#3.cram  1204-PF-GM-CP1  23384.0       1.0        3.0   
/seq/23384/23384_1#4.cram             NaN  23384.0       1.0        4.0   
/seq/23384/23384_1#5.cram  1204-PF-GM-CP1  23384.0       1.0        5.0   

                                   qc_complete  manual_qc  \
irods_path                                                  
/seq/23384/23384_1#1.cram  2017-08-21 14:30:49        1.0   
/seq/23384/23384_1#2.cram  2017-08-21 14:30:49        1.0   
/seq/23384/23384_1#3.cram  2017-08-21 14:30:49        1.0   
/seq/23384/23384_1#4.cram  2017-08-21 14:30:49        1.0   
/seq/23384/23384_1#5.cram  2017-08-21 14:30:49        1.0   

                                   description instrument_name  \
irods_path                                                       
/seq/23384/23384_1#1.cram  2017-08-21 14:30:49            HX10   
/seq/23384/23384_1#2.cram  2017-08-21 14:30:49            HX10   
/seq/23384/23384_1#3.cram  2017-08-21 14:30:49            HX10   
/seq/23384/23384_1#4.cram  2017-08-21 14:30:49            HX10   
/seq/23384/23384_1#5.cram  2017-08-21 14:30:49            HX10   

                          instrument_model  forward_read_length  \
irods_path                                                        
/seq/23384/23384_1#1.cram           HiSeqX                150.0   
/seq/23384/23384_1#2.cram           HiSeqX                150.0   
/seq/23384/23384_1#3.cram           HiSeqX                150.0   
/seq/23384/23384_1#4.cram           HiSeqX                150.0   
/seq/23384/23384_1#5.cram           HiSeqX                150.0   

                           requested_insert_size_from  \
irods_path                                              
/seq/23384/23384_1#1.cram                       450.0   
/seq/23384/23384_1#2.cram                       450.0   
/seq/23384/23384_1#3.cram                       450.0   
/seq/23384/23384_1#4.cram                       450.0   
/seq/23384/23384_1#5.cram                       450.0   

                           requested_insert_size_to human_percent_mapped  \
irods_path                                                                 
/seq/23384/23384_1#1.cram                     450.0           

In [20]:
df_magnus_not_richard['study_name'].value_counts(dropna=False)

1176-PF-KE-CP1                                        2458
1204-PF-GM-CP1                                        2310
Malaria Programme R&D                                 2192
NaN                                                    824
Team 112 R&D                                           594
Malaria R&D                                            473
AT-rich amplification                                   18
1083-PF-GH-CONWAY                                       12
IHTP_PWGS 1175-VO-KH-STLAURENT                          12
ILB_1044-PF-KH-FAIRHURST                                12
ILB_1126-PF-LAB-FAIRHURST                                8
1148-PF-BD-MAUDE                                         6
Test sequencing of high level human contamination.       6
ILB_1106-PV-MULTI-PRICE_INDONESIA                        1
Name: study_name, dtype: int64

In [21]:
df_magnus['study'].value_counts(dropna=False).head(10)

NaN                        17488
1052-PF-TRAC-WHITE          1052
1125-PF-TH-NOSTEN            674
1040-PF-GB-HAMILTON          636
1022-PF-MW-OCHOLLA           318
1095-PF-TZ-ISHENGOMA         300
1044-PF-KH-FAIRHURST         288
1094-PF-GH-AMENGA-ETEGO      256
1093-PF-CM-APINJOH           239
1008-PF-SEA-RINGWALD         234
Name: study, dtype: int64

# Conclusion
There are multiple problems with this version of the Pf6.2 manifest from fits:
- A small number of lanelets have multiple rows, with one version containing "no_cal" in the file path
- Most lanelets have no study associated. This is required for all rows
- There are many lanelets from studies which should have been excluded, e.g. 1176-PF-KE-CP1, 1204-PF-GM-CP1 and R&D studies
- Most worryingly, there are a large number of lanelets that are missing but that were included in previous builds. In Richard's version of the manifest most, but not all of these were missing sanger_sample_id. I couldn't spot any other obvious pattern.

Note that I used a file to map sequencescape studies to Alfresco studies. This can be found in ```/nfs/team112_internal/rp7/src/github/wtsi-team112/SIMS/meta/mlwh/sequencescape_alfresco_study_mappings.txt``` and was created with https://github.com/wtsi-team112/SIMS/blob/master/notebooks/rp7/20180203_mlwh_exceptions_files.ipynb
